src/Nethermind/Nethermind.Db/DbExtensions.cs

```diff
@@ -131,10 +131,15 @@ public static void Delete(this IDb db, long key)
                 if (db is IDbWithSpan spanDb && decoder is IRlpValueDecoder<TItem> valueDecoder)
                 {
                     Span<byte> data = spanDb.GetSpan(key);
-                    if (data.IsNullOrEmpty())
+                    if (data.IsNull())
                     {
                         return null;
                     }
+                    else if (data.Length == 0)
```

Comment:
can you move this if under 'try'? Then we have one release in `finally`.

Reply:
Done